dts/arm/infineon/xmc4500.dtsi

```diff
@@ -23,3 +24,152 @@
 &flash0 {
 	reg = <0xc000000 DT_SIZE_M(1)>;
 };
+
+&pinctrl {
+	ranges = <0x48028000 0x48028000 0x700>;
+
+	gpio3: gpio@48028300 {
+		compatible = "infineon,xmc4xxx-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x48028300 0x100>;
+		label = "GPIO3";
+		status = "disabled";
+	};
+
+	gpio4: gpio@48028400 {
+		compatible = "infineon,xmc4xxx-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x48028400 0x100>;
+		label = "GPIO4";
+		status = "disabled";
+	};
+
+	gpio5: gpio@48028500 {
+		compatible = "infineon,xmc4xxx-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x48028500 0x100>;
+		label = "GPIO5";
+		status = "disabled";
+	};
+
+	gpio6: gpio@48028600 {
+		compatible = "infineon,xmc4xxx-gpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+		reg = <0x48028600 0x100>;
+		label = "GPIO6";
+		status = "disabled";
+	};
+};
+
+&pinctrl {
+	uart_tx_p0_1_u1c1: uart_tx_p0_1_u1c1 {
+		pinmux = <XMC4XXX_PINMUX_SET(0, 1, 2)>;
+		drive-strength = "strong-soft-edge";
+		drive-push-pull;
+		output-high;
+	};
+	uart_tx_p0_5_u1c0: uart_tx_p0_5_u1c0 {
+		pinmux = <XMC4XXX_PINMUX_SET(0, 5, 2)>;
+		drive-strength = "strong-soft-edge";
+		drive-push-pull;
+		output-high;
+	};
+	uart_tx_p1_5_u0c0: uart_tx_p1_5_u0c0 {
+		pinmux = <XMC4XXX_PINMUX_SET(1, 5, 2)>;
+		drive-strength = "strong-soft-edge";
+		drive-push-pull;
+		output-high;
+	};
```

Comment:
- predefined nodes should be in a separate header ideally, e.g. `xmc4500-pinctrl.dtsi`. Also, options are tipically package dependent (not SoC only). I see 100/144 pin packages for this SoC.
- Defaults should be avoided in general, except there's one single valid combination. Please justify.

Reply:
> * predefined nodes should be in a separate header ideally, e.g. `xmc4500-pinctrl.dtsi`. Also, options are tipically package dependent (not SoC only). I see 100/144 pin packages for this SoC.

I'll put the pre-defined nodes in a separate file and use a name with a pin package. Btw, I also have a pull request to rename xmc4500.dtsi:
https://github.com/zephyrproject-rtos/zephyr/pull/47161/commits/e7c0d85526166b7716a51e4d640f87fd344e731d

Reply:
> * Defaults should be avoided in general, except there's one single valid combination. Please justify.

I'll move the pin config settings to the board dts.